### ISSUE TITLE
`prefer-single-call` can cause breakings

### ISSUE THREAD

**Issue by rafipiccolo** [sindresorhus/eslint-plugin-unicorn#2636]
i dont know if its exactly a bug, but i dont believe this rule should be activated by default.
or maybe it should only handle arrays because issues arrise when merging pushes on "new stream.Readable();"

prefer-single-call

consider this custom execFilePromise function.
it works with 3 separate pushes but not with merged pushes.

```js
import { execFile } from 'node:child_process';
import stream from 'node:stream';

function execFilePromise(cmd, args, options) {
    return new Promise((resolve, reject) => {
        const child = execFile(cmd, args, options, (err, stdout, stderr) => {
            const code = err?.code ?? 0;
            const ignoreExitCode = options?.ignoreExitCode ?? 0;

            if (err && !ignoreExitCode) return reject(err);

            resolve({ stdout, stderr, code });
        });

        if (options?.stdin) {
            const stdinStream = new stream.Readable();
            stdinStream.push(options.stdin);
            stdinStream.push(null);

            // this gets replaced to stdinStream.push(options.stdin, null);
            // which throws errors "_read method not implemented"

            stdinStream.pipe(child.stdin);
        }
    });
}

export default execFilePromise;

```

**Comment by fisker** [comment]
This discussed in #1057, can't avoid false positives without type information.
